### PR TITLE
[azure-iot-sdk-c] LTS release 2022-01-21

### PR DESCRIPTION
**Pull request by momuno**
- #### What does your PR fix?  
LTS release for 2022-01-21

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
Unchanged

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by momuno** [comment]
Hi @JonLiu1993 .  Ready for merge. Thanks!

**Comment by dg0yt** [comment]
Does this update resolve #22777?

I don't know the canonical form of using this port in a cmake user project, but I feel like the auto-generated usage after `vcpkg install azure-iot-sdk-c` should be replaced by an explicit `usage` file.

File: ports/azure-iot-sdk-c/portfile.cmake [hunk 1]
```diff
@@ -8,20 +8,20 @@ if("public-preview" IN_LIST FEATURES)
         SHA512 6798b17d6768b3ccbd0eb66719b50f364cd951736eb71110e2dc9deca054a1566ff88b9e8c5e9b52536e4308cad6cd3cbebff3282c123083e3afaee5535e724b
         HEAD_REF public-preview
         PATCHES
+            fix-install-location.patch
             improve-external-deps.patch
-            fix-cmake.patch
```
Comment (dg0yt) [inline ports/azure-iot-sdk-c/portfile.cmake:12]: I'm glad that this patch is gone. But AFAICT from looking at this PR in GH, it still needs to be removed with `git rm`.

Comment (dg0yt) [reply ports/azure-iot-sdk-c/portfile.cmake:12]: Thanks!

**Comment by momuno** [comment]
> Does this update resolve #22777?
> 
> I don't know the canonical form of using this port in a cmake user project, but I feel like the auto-generated usage after `vcpkg install azure-iot-sdk-c` should be replaced by an explicit `usage` file.

Investigating. Thanks for the heads up.

**Review by JonLiu1993 — APPROVED** [review]

File: ports/azure-iot-sdk-c/improve-external-deps.patch [hunk 2]
```diff
@@ -100,18 +46,25 @@ index d05d654..5872c8d 100644
...
- setSdkTargetBuildProperties(iothub_service_client)
+diff --git a/deps/CMakeLists.txt b/deps/CMakeLists.txt
+index 22e24f851..7dc30beba 100644
+--- a/deps/CMakeLists.txt
++++ b/deps/CMakeLists.txt
+@@ -37,5 +37,4 @@ endif()
+ if(NOT ${use_installed_dependencies})
+     add_subdirectory(${CMAKE_CURRENT_LIST_DIR}/parson)
+ else()
+-    find_package(parson REQUIRED CONFIG)
```
Comment (ras0219-msft) [inline ports/azure-iot-sdk-c/improve-external-deps.patch:57]: If `parson` is now deploying config files, we should update `parson` so we can avoid this patch.

**Comment by ras0219-msft** [comment]
Thanks for the update!

File: ports/azure-iot-sdk-c/improve-external-deps.patch (hunk 2 above)
Comment (momuno) [reply ports/azure-iot-sdk-c/improve-external-deps.patch:57]: Hi @ras0219-msft.  I have created a task for my team for us to follow up on these patches.  Thanks.

**Pull request merged**